collector/config.go

```diff
@@ -60,15 +80,67 @@ func (c Config) Validate() error {
 	return nil
 }
 
+// decryptPrivateKey returns a RSA private key from the PrivateKeyPath and PrivateKeyPassword fields
+// of the config.
+// Assumes that the private key is encrypted in PKCS #8 syntax, as is recommended by Snowflake
+func (c Config) decryptPrivateKey() (*rsa.PrivateKey, error) {
+	var parsedPrivateKey *rsa.PrivateKey
+	pk, err := os.ReadFile(c.PrivateKeyPath)
+	if err != nil {
+		fmt.Printf("Error opening file: %s", err)
+		return nil, err
```

Comment:
@Caleb-Hurshman Sorry I should have been a little more explicit. Don't actually log an error here (it will get logged after it bubbles up). But return the nicer wrapped error with `fmt.Errorf`